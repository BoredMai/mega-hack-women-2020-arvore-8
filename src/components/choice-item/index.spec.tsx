import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import ChoiceItem from '.';

describe('<ChoiceItem>', () => {
  const choice = { icon: 'princesa', label: 'Princesa', url: 'princesa' };
  const props = {
    choice,
    isSelected: false,
    onSelect: jest.fn().mockName('onSelect'),
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<ChoiceItem {...props} />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('adds class when choice is disabled', () => {
    wrapper.setProps({ choice: { ...choice, disabled: true } });
    expect(wrapper.find('.choiceDisabled')).toHaveLength(1);
  });

  it('adds class when choice is selected', () => {
    wrapper.setProps({ isSelected: true });
    expect(wrapper.find('.choiceSelected')).toHaveLength(1);
  });

  it('calls onSelect on button click', () => {
    wrapper.find('.choice').simulate('click');

    expect(props.onSelect).toHaveBeenCalledWith(choice);
  });
});
