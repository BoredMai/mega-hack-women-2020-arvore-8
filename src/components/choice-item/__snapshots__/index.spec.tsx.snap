// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChoiceItem> renders properly 1`] = `
<div
  className="choice"
  onClick={[Function]}
>
  <img
    id="user-choice-Princesa"
    src="src/assets/princesa"
  />
  <label
    htmlFor="user-choice-Princesa"
  >
    Princesa
  </label>
</div>
`;
